### ISSUE TITLE
Crash in data view when column name contains “$”

### ISSUE THREAD

**Issue by BlueM** [Sequel-Ace/Sequel-Ace#1648]
Sequel Ace 4.0.3, build 20042 on macOS Ventura / 13.0.1 on a 14" MacBook Pro (Apple M1 Pro).

On one of the servers with which I use Sequel Ace, one of the views crashed reproducible. This view used MySQL’s `json_extract()` method for one column, and as I didn’t specify an alias, the effective name of the column was `json_extract(columnname, '$.message')`. After some tests, it was obvious that Sequel Ace does not like the dollar sign ($)   in column names – both in tables or views.

This means, you will get a reproducible crash when viewing the data for the following view or table:

```sql
CREATE VIEW testview AS SELECT 'abc' AS `$abc`;
CREATE TABLE testtable (`ab$c` int);
```

**Comment by DannyJJK** [comment]
I am able to replicate this on 4.0.4. Looks like there's a regex parser and the $ symbol breaks it.

Here's a part of the log from running Sequel Ace in debug mode when it crashes

```
2022-12-21 19:49:53.279118+0000 Sequel Ace[12865:688188] -[NSMutableString(RegexKitLiteAdditions) replaceOccurrencesOfRegex:withString:]:2621: replacement: `$test`
2022-12-21 19:49:53.279305+0000 Sequel Ace[12865:688188] NSException *rkl_NSExceptionForRegex(NSString *, RKLRegexOptions, const UParseError *, int32_t):1739: RKLICURegexException. Reason: ICU regular expression error #66325, U_REGEX_INVALID_CAPTURE_GROUP_NAME.. userInfo: {
    NSLocalizedFailureReason = "The error U_REGEX_INVALID_CAPTURE_GROUP_NAME occurred.";
    RKLICURegexErrorCode = 66325;
    RKLICURegexErrorName = "U_REGEX_INVALID_CAPTURE_GROUP_NAME";
    RKLICURegexRegex = "(?<!\\\\)\\$CURRENT_FIELD";
    RKLICURegexRegexOptions = 0;
}
```

**Comment by Jason-Morcos** [comment]
Thanks for this investigation, @DannyJJK! That helped a lot in fixing this issue. PR incoming now

**Comment by Jason-Morcos** [comment]
Please try the latest beta and confirm it's all good - the fix for this issue is included in the latest 4.0.5+ beta!

https://github.com/Sequel-Ace/Sequel-Ace/releases

**Comment by DannyJJK** [comment]
@Jason-Morcos Yep it's all good now:

<img width="490" alt="image" src="https://user-images.githubusercontent.com/11057501/210138673-ca31342c-e085-4a8e-a988-5ce1c58e6fe5.png">

This previously would have crashed Sequel Ace on 4.0.4, but on the 4.0.5 beta it displays correctly.
